pangeo_forge_runner/bakery/flink.py

```diff
@@ -192,6 +280,12 @@ def make_flink_deployment(self, name: str, worker_image: str):
             },
         }
 
+        if self.enable_job_archiving:
+            flink_deploy["spec"]["jobManager"].update(
+                self._job_manager_pod_template_with_archiving()
```

Comment:
my preference here is that we instead have a `add_job_manager_template` function, to which we can pass the `flink_deploy` object, and it does whatever changes needed and returns it, rather than use `update`.